### PR TITLE
plan: adjust index join's cost estimation model

### PR DESCRIPTION
**Pull request by lamxTyler**
PTAL @hanfei1991  @winoros

### CODE REVIEW

File: plan/task.go [hunk 1]
```diff
@@ -142,7 +142,12 @@ func (p *PhysicalIndexJoin) attach2Task(tasks ...task) task {
 }
 
 func (p *PhysicalIndexJoin) getCost(lCnt float64) float64 {
+	if lCnt <= 1 {
+		lCnt = 1
+	}
 	cst := lCnt * netWorkStartFactor
+	batchSize := p.ctx.GetSessionVars().IndexLookupSize
+	cst += lCnt * math.Log2(math.Min(float64(batchSize), lCnt)) * 2
```
Comment (hanfei1991) [inline plan/task.go]: use p.batchSize

Comment (lamxTyler) [inline plan/task.go]: It does not have such member?

Comment (hanfei1991) [inline plan/task.go]: We, I'm wrong. If the `KeepOrder` is true, it's size is 1.

**Comment by hanfei1991** [comment]
LGTM

**Comment by hanfei1991** [comment]
@winoros PTAL

File: plan/task.go [hunk 2]
```diff
@@ -142,7 +142,16 @@ func (p *PhysicalIndexJoin) attach2Task(tasks ...task) task {
 }
 
 func (p *PhysicalIndexJoin) getCost(lCnt float64) float64 {
-	cst := lCnt * netWorkStartFactor
+	if lCnt <= 1 {
```
Comment (winoros) [inline plan/task.go]: `<` is enough.

File: plan/task.go [hunk 3]
```diff
@@ -142,7 +142,16 @@ func (p *PhysicalIndexJoin) attach2Task(tasks ...task) task {
...
+		lCnt = 1
+	}
+	cst := lCnt * netWorkFactor
+	batchSize := p.ctx.GetSessionVars().IndexLookupSize
```
Comment (winoros) [inline plan/task.go]: Store it as a variable of `PhysicalIndexJoin`? This is also used when build IndexJoin Executor.

**Pull request merged**